bfit/fit.py

```diff
@@ -514,9 +542,54 @@ def run(self, c0, e0, opt_coeffs=True, opt_expons=True, maxiter=1000, tol=1.e-14
         callback = None
         if disp:
             if self.method == "slsqp":
-                callback = lambda xk : print(self.goodness_of_fit(xk[:len(c0)], xk[len(c0):]))
+                # Template for the header.
+                print("-" * (10 + 12 + 15 + 15 + 15 + 12))
+                # Format is {identifier:width}, ^ means center it, | means put a bar in it
+                template_header = (
+                    "|{0:^12}|{1:^15}|{2:^15}|{3:^15}|{4:^16}|"
+                )
+                # Print the headers
+                print(template_header.format(
+                    "Integration", "L1", "L Infinity", "Least-squares", "Kullback-Leibler")
+                )
+                print("-" * (10 + 12 + 15 + 15 + 15 + 12))
+                # Template for float iteration
+                template_iters = (
+                    "|{0:^12f}|{1:^15e}|{2:^15f}|{3:^15e}|{4:^16e}|"
+                )
+
+                def print_results(xk):
+                    if opt_coeffs and opt_expons:
+                        performance = self.goodness_of_fit(xk[:len(c0)], xk[len(c0):])
+                    elif opt_coeffs:
+                        performance = self.goodness_of_fit(xk, e0)
+                    elif opt_expons:
+                        performance = self.goodness_of_fit(c0, xk)
+                    print(template_iters.format(
+                        performance[0], performance[1], performance[2],
+                        performance[3], performance[4])
+                    )
```

Comment:
I think `print(template_iters.format(*performance)` should work. @Ali-Tehrani can you please try it?

Reply:
I does work.  I changed it in all three places and commited it.